redfish-core/lib/log_services.hpp

```diff
@@ -1475,6 +1522,230 @@ inline void requestRoutesJournalEventLogEntry(App& app)
             });
 }
 
+template <typename Callback>
+void getHiddenPropertyValue(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& entryId, Callback&& callback)
+{
+    auto respHandler = [callback{std::move(callback)},
+                        asyncResp](const boost::system::error_code ec,
+                                   const std::variant<bool>& hiddenProperty) {
+        bool hiddenPropVal = false;
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "DBUS response error: " << ec;
+
+            // Error while retrieving Hidden Property. This can happen
+            // when the property is not found in the event log entry. In such
+            // cases try to process the entry as if it is a normal entry
+            // with Hidden property set to false
+            callback(hiddenPropVal);
+            return;
+        }
+        hiddenPropVal = std::get<bool>(hiddenProperty);
+        callback(hiddenPropVal);
+    };
+
+    // Get the Hidden Property
+    crow::connections::systemBus->async_method_call(
+        respHandler, "xyz.openbmc_project.Logging",
+        "/xyz/openbmc_project/logging/entry/" + entryId,
+        "org.freedesktop.DBus.Properties", "Get",
+        "org.open_power.Logging.PEL.Entry", "Hidden");
+}
+enum eventLogTypes
+{
+    eventLog = 1,
+    ceLog,
+};
+
+inline void getDBusLogEntryCollection(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    GetManagedObjectsType& resp, eventLogTypes type)
+{
+    nlohmann::json& entriesArray = asyncResp->res.jsonValue["Members"];
+    entriesArray = nlohmann::json::array();
+    for (auto& objectPath : resp)
+    {
+        uint32_t* id = nullptr;
+        std::time_t timestamp{};
+        std::time_t updateTimestamp{};
+        std::string* severity = nullptr;
+        std::string* message = nullptr;
+        std::string* filePath = nullptr;
+        bool resolved = false;
+        bool hiddenPropVal = false;
+        for (auto& interfaceMap : objectPath.second)
+        {
+            if (interfaceMap.first == "xyz.openbmc_project.Logging.Entry")
+            {
+                for (auto& propertyMap : interfaceMap.second)
+                {
+                    if (propertyMap.first == "Id")
+                    {
+                        id = std::get_if<uint32_t>(&propertyMap.second);
+                        if (id == nullptr)
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
```

Comment:
Added nullptr checks for Id, Severity and Message properties similar to Resolved property. I think returning an error is not correct - it should simply break out of the loop to process the next event log entry this this service is to list all entries?
If it is an issue, this is there in upstream code as well...

@spinler @gtmills any thoughts?

Reply:
Leaving the code as is since that is the way it is in upstream...